src/main/java/com/iota/iri/service/snapshot/conditions/SnapshotSizeCondition.java

```diff
@@ -0,0 +1,105 @@
+     */
+    private static final int MILESTONES = 5;
+    
+    /**
+     * The maximum size we want the DB to have, with margin due to DB delays for writing to disk
+     */
+    private final long maxSize;
+    private final Tangle tangle;
+
+    private final SnapshotProvider snapshotProvider;
+    
+    /**
+     * The cached size of the db, so we don't keep snapshotting when the size doesn't change after a snapshot 
+     */
+    private long lastSize = -1;
+
+    private SnapshotConfig config;
+
+    /**
+     * Implements a {@link SnapshotCondition} based on the total database size
+     * 
+     * @param tangle the database interface.
+     * @param config configuration with snapshot specific settings.
+     * @param snapshotProvider gives us access to the relevant snapshots.
+     */
+    public SnapshotSizeCondition(Tangle tangle, SnapshotConfig config, SnapshotProvider snapshotProvider) {
+        this.tangle = tangle;
+        this.config = config;
+        if (config.getLocalSnapshotsPruningEnabled()) {
+            maxSize = (long) Math.floor(IotaUtils.parseFileSize(config.getLocalSnapshotsDbMaxSize()) / 100 * (100-MARGIN));
+            
+            if (config.getLocalSnapshotsPruningDelay() != BaseIotaConfig.Defaults.LOCAL_SNAPSHOTS_PRUNING_DELAY_MIN) {
+                log.warn("We recommend setting pruning delay to the minimum(" 
+                        + BaseIotaConfig.Defaults.LOCAL_SNAPSHOTS_PRUNING_DELAY_MIN 
+                        + ") when using db size limitation.");
+            }
+        } else {
+            if (config.getLocalSnapshotsDbMaxSize() != BaseIotaConfig.Defaults.LOCAL_SNAPSHOTS_DB_MAX_SIZE) {
+                log.warn("Local snapshots with size condition does not work with pruning disabled");
+            }
+            maxSize = -1;
+        }
+        this.snapshotProvider = snapshotProvider;
+    }
+
+    /**
+     * We only take a snapshot if the db size updated since last check and the new size is bigger than the maximum allowed.
+     * A margin of 1% is taken, based on the min amount of milestones, resulting in ~10GB DB, 100MB margin.
+     * This corresponds to the, default, 64MB buffer for writing to disk, which is at most 128 of size.
+     * 
+      * {@inheritDoc}
+     */
+    @Override
+    public boolean shouldTakeSnapshot(boolean isInSync) {
+        long size = tangle.getPersistanceSize();
+        if (size == lastSize || maxSize <= 0) {
+            return false;
+        } 
+        
+        // Update the last size
+        lastSize = size;
+        
+        return size > maxSize;
+    }
+
+    @Override
+    public int getSnapshotStartingMilestone() throws SnapshotException {
+        // Snapshot by size doesn't need a recent snapshot, we calculate here so we can reuse it when we call getSnapshotPruningMilestone.
+        // This will not go below the allowed snapshot depth due to the other conditions
+        return -1;
```

Comment:
Aha, yes but the size condition is indeed only for pruning, therefor doesnt need any snapshots except making sure we have a snapshot taken over the data we prune.
Doesnt have to be the same time, but it must be snapshotted otherwise we wont have a solidentrypoint